k9mail/src/main/java/com/fsck/k9/controller/MessagingController.java

```diff
@@ -726,899 +714,65 @@ public void run() {
     /**
      * Start foreground synchronization of the specified folder. This is generally only called
      * by synchronizeMailbox.
-     * <p>
-     * TODO Break this method up into smaller chunks.
      */
     @VisibleForTesting
-    void synchronizeMailboxSynchronous(final Account account, final String folder, final MessagingListener listener,
+    void synchronizeMailboxSynchronous(final Account account, final String folderName, final MessagingListener listener,
             Folder providedRemoteFolder) {
-        Folder remoteFolder = null;
-        LocalFolder tLocalFolder = null;
 
-        Timber.i("Synchronizing folder %s:%s", account.getDescription(), folder);
+        /*
+         * Synchronization process:
+         *
+        Open the folder
+        Upload any local messages that are marked as PENDING_UPLOAD (Drafts, Sent, Trash)
+        Get the message count
+        Get the list of the newest K9.DEFAULT_VISIBLE_LIMIT messages
+        getMessages(messageCount - K9.DEFAULT_VISIBLE_LIMIT, messageCount)
+        See if we have each message locally, if not fetch it's flags and envelope
+        Get and update the unread count for the folder
+        Update the remote flags of any messages we have locally with an internal date newer than the remote message.
+        Get the current flags for any messages we have locally but did not just download
+        Update local flags
+        For any message we have locally but not remotely, delete the local message to keep cache clean.
+        Download larger parts of any new messages.
+        (Optional) Download small attachments in the background.
+         */
+
+        Timber.i("Synchronizing folder %s:%s", account.getDescription(), folderName);
 
         for (MessagingListener l : getListeners(listener)) {
-            l.synchronizeMailboxStarted(account, folder);
+            l.synchronizeMailboxStarted(account, folderName);
```

Comment:
I think we should try to keep these pairs of listener callbacks in the same method where possible. in this case, just move them to the beginning of the performSync methods, and also into the "don't ever sync outbox or errors folder" statement